buildSrc/src/main/kotlin/terasology-module.gradle.kts

```diff
@@ -0,0 +1,246 @@
+val moduleConfig: Map<String, Any> = slurper.parseText(moduleFile.readText())!! as Map<String, Any>
+for (dependency in moduleConfig["dependencies"] as List<Map<String, String>>) {
+    if (dependency["id"] != "engine") {
+        moduleDepends.add(dependency["id"]!!)
+    }
+}
+
+// Gradle uses the magic version variable when creating the jar name (unless explicitly set somewhere else I guess)
+version = moduleConfig["version"] as String
+
+// Check for an outright -SNAPSHOT in the loaded version - for ease of use we want to get rid of that everywhere, so warn about it and remove for the variable
+val undesiredSnapshotTag = version.toString().endsWith("-SNAPSHOT")
+if (undesiredSnapshotTag) {
+    println("Taking off undesired -SNAPSHOT")
+    version = version.toString().removeSuffix("-SNAPSHOT")
+    println("WARNING: Module ${project.name} is explicitly versioned as a snapshot in module.txt, please remove '-SNAPSHOT'")
+}
+
+// The only case in which we make module non-snapshots is if release management is enabled and BRANCH_NAME is "master"
+if (moduleConfig["isReleaseManaged"].toString().toBoolean() && env["BRANCH_NAME"] == "master") {
+    // This is mildly awkward since we need to bypass by default, yet if release management is on (true) then we set the bypass to false ..
+    val bypassModuleReleaseManagement by extra(false)
+} else {
+    // In the case where we actually are building a snapshot we load -SNAPSHOT into the version variable, even if it wasn't there in module.txt
+    version = "${version}-SNAPSHOT"
+}
+
+// Jenkins-Artifactory integration catches on to this as part of the Maven-type descriptor
+group = "org.terasology.modules"
+
+println("Version for $project.name loaded as $version for group $group")
+
+// Grab all the common stuff like plugins to use, artifact repositories, code analysis config, Artifactory settings, Git magic
+// Note that this statement is down a ways because it is affected by the stuff higher in this file like setting versioning
+apply(from = "$rootDir/config/gradle/publish.gradle")
+
+// Handle some logic related to where what is
+configure<SourceSetContainer> {
+    named("main") {
+        java.outputDir = File("$buildDir/classes")
+    }
+    named("test") {
+        java.outputDir = File("$buildDir/testClasses")
+    }
+}
+val convention = project.getConvention().getPlugin(JavaPluginConvention::class)
+val mainSourceSet = convention.getSourceSets().getByName("main");
+
+// TODO: Remove when we don't need to rely on snapshots. Needed here for solo builds in Jenkins
+configurations.all {
+    resolutionStrategy.cacheChangingModulesFor(0, "seconds")
+}
+
+// Set dependencies. Note that the dependency information from module.txt is used for other Terasology modules
+dependencies {
+    // Check to see if this module is not the root Gradle project - if so we are in a multi-project workspace
+    "implementation"(group = "org.terasology.engine", name = "engine", version = "+") { isChanging = true }
```

Comment:
you have extra quotes :D

Reply:
Cerv asked about that too. I tried taking them out, and gradle seems just fine with that, but IDEA's gradle sync broke on me at that point. :disappointed:  I asked about it on the [gradle forum](https://discuss.gradle.org/t/buildsrc-precompiled-kotlin-script-problems-in-intellij/36249?u=keturn), I'll report back if I hear anything.

Reply:
hmm. It is worked for me :/

Reply:
Oh, that's good to know.

I hit "Invalidate Caches and Restart" and after a rebuild and a bit that part works here too. :+1:

Reply:
We need hit this button often, after `.idea` PR :(